### PR TITLE
0.13 implement template string tests

### LINKED ISSUE #3698: Implement new tests in `template-string.ts`

### PR DESCRIPTION
**Pull request by vvagaytsev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Partially addresses #3698

**Special notes for your reviewer**:

### CODE REVIEW

File: core/test/unit/src/template-string.ts [hunk 1]
```diff
@@ -1710,58 +1710,176 @@ describe("collectTemplateReferences", () => {
...
+      expectError(() => getActionTemplateReferences(config), {
+        contains: "Found invalid action reference (name is not a string)",
+      })
+    })
 
-  it("throws if runtime ref has no name", () => {
-    throw "TODO"
+    it("throws if action name is not resolvable", () => {
+      throw "TODO"
+    })
```
Comment (vvagaytsev) [inline core/test/unit/src/template-string.ts:1802]: This has been left as `throws "TODO"`. There is no validation for non-resolvable names yet. Every name is just considered as a string.

@edvald is this related to the [`TODO-G2: Allow such nested references in certain cases, e.g. if resolvable with a ProjectConfigContext.` comment](https://github.com/garden-io/garden/blob/0.13/core/src/template-string/template-string.ts#L429) in [template-string.ts](https://github.com/garden-io/garden/blob/0.13/core/src/template-string/template-string.ts)?

File: core/test/unit/src/template-string.ts [hunk 2]
```diff
@@ -1710,58 +1710,176 @@ describe("collectTemplateReferences", () => {
...
+        foo: '${runtime["tasks"].123}',
+      }
+      expectError(() => getActionTemplateReferences(config), {
+        contains: "Found invalid runtime reference (name is not a string)",
+      })
+    })
+
+    it("throws if runtime ref name is not resolvable", () => {
+      throw "TODO"
+    })
```
Comment (vvagaytsev) [inline core/test/unit/src/template-string.ts:1882]: The same question as [above](https://github.com/garden-io/garden/pull/3699/files#r1098647747).

File: core/test/unit/src/template-string.ts [hunk 3]
```diff
@@ -1718,42 +1718,85 @@ describe("getActionTemplateReferences", () => {
...
+      foo: "${action[123]}",
+    }
+    expectError(() => getActionTemplateReferences(config), {
+      contains: "Found invalid action reference (kind is not a string)",
+    })
+  })
+
+  it("throws if action ref has no name", () => {
+    const config = {
+      foo: "${action[\"build\"]}",
```
Comment (Orzelius) [inline core/test/unit/src/template-string.ts]: maybe just using ` would be easier to parse here or does the lint want it this way?
like this:
```
foo: `${action["build"]}`
```

Comment (Orzelius) [reply core/test/unit/src/template-string.ts]: I think `'` would also work the same

**Review by Orzelius — APPROVED** [review]
Thank you!

File: core/test/unit/src/template-string.ts (hunk 1 above)
Comment (vvagaytsev) [reply core/test/unit/src/template-string.ts:1802]: Leaving this discussion open and merging the PR. The remained TODO-tests will be implemented in a separate PR.

Comment (edvald) [reply core/test/unit/src/template-string.ts:1802]: I think this can be removed

File: core/test/unit/src/template-string.ts (hunk 2 above)
Comment (edvald) [reply core/test/unit/src/template-string.ts:1882]: Same as above, just remove

**Pull request merged**